x-pack/plugins/file_data_visualizer/public/plugin.ts

```diff
@@ -40,7 +43,11 @@ export class FileDataVisualizerPlugin
       FileDataVisualizerSetupDependencies,
       FileDataVisualizerStartDependencies
     > {
-  public setup() {}
+  public setup(core: CoreSetup, plugins: FileDataVisualizerSetupDependencies) {
+    if (plugins.home) {
```

Comment:
Should there be a permission check here to only register the file upload tab if the user has the minimum permissions needed to analyze a file?

Would this be a good time to remove `access:fileUpload:analyzeFile` and `access:fileUpload:import` tags from file upload routes?

Reply:
I think this would be a good time to decouple file data visualizer permissions from ML so that users can upload a file and not  have kibana access to ml or any ml roles.

Reply:
> Should there be a permission check here to only register the file upload tab if the user has the minimum permissions needed to analyze a file?

ML has run in to race condition issues before when registering links in home and the management app where, due to the time it takes to check the license and in this case maybe the capabilities, the registration happens too late for the links to actually appear.

I'll take a look to see if this is possible, but we might just have to always register and perform the capabilities check later when the user navigates to the "Upload file" tab.

> Would this be a good time to remove access:fileUpload:analyzeFile and access:fileUpload:import tags from file upload routes?

I might have missed something, why would we want to remove the capabilities checks from the routes?